ros/resources/v0/hosts.py

```diff
@@ -0,0 +1,67 @@
+import json
+from flask import jsonify
+from flask import request, Response
+from flask_restful import Resource, abort, fields, marshal_with
+from marshmallow import Schema, fields
+
+from ros.models import PerformanceProfile
+from ros.utils import is_valid_uuid
+from ros.lib.host_inventory_interface import fetch_all_hosts_from_inventory
+
+import logging
+
+class PerformanceProfileSchema(Schema):
+    inventory_id = fields.Str()
+    performance_record = fields.Dict(keys=fields.Str(), values=fields.Str())
+    performance_score = fields.Dict(keys=fields.Str(), values=fields.Int())
+
+class HostsApi(Resource):
+    def get(self):
+        auth_key = request.headers.get('X-RH-IDENTITY')
+        #logging.info(auth_key)
+        inv_hosts = fetch_all_hosts_from_inventory(auth_key)
+
+        pp_schema = PerformanceProfileSchema(many=True)
+        profile_result = pp_schema.dump(PerformanceProfile.query.all())
+        p_score_traversed = 0
+
+        # match the performance_profile with the host
+
+        for i in range(0, len(inv_hosts['results'])):
+            for j in range (0, len(profile_result)):
+                if  inv_hosts['results'][i]['id'] != profile_result[j]['inventory_id']:
+                    continue
+                else:
+                    inv_hosts['results'][i]['performance_score'] = profile_result[j]['performance_score']
+                    p_score_traversed += 1
+                    break
+            if p_score_traversed == len(profile_result):
+                break
+        # Removing these elements temporarily in order to match result with dummy_systems
+        inv_hosts.pop('total')
+        inv_hosts.pop('count')
+        inv_hosts.pop('page')
+        inv_hosts.pop('per_page')
+
+        # Adding these fields as currently not present in rosdb
+        inv_hosts['results'][0]['recommendation_count'] = 5
+        inv_hosts['results'][0]['state'] = 'Crashloop'
+        print (inv_hosts)
+
+        return jsonify(inv_hosts)
```

Comment:
```suggestion
       inv_host_ids = [host['id'] for host in inv_hosts['results']]
        profile_result = PerformanceProfile.query.filter(PerformanceProfile.inventory_id.in_(inv_host_ids)).order_by(PerformanceProfile.report_date.desc()).all()
        
        hosts = []
        for i in profile_result:
            if len(list(filter(lambda host: host['id'] == str(i.inventory_id), hosts))):
                continue
            else:
                host = list(filter(lambda host: host['id'] == str(i.inventory_id), inv_hosts['results']))[0]
                host['performance_score'] = i.__dict__['performance_score']
                host['recommendation_count'] = 5
                host['state'] = 'Crashloop'
                hosts.append(host)

        return {"results": hosts}
```

Reply:
@patilsuraj767 Wouldn't it be better to filter all inv_hosts whose ids match with profile.inventory_id instead of filtering profiles based on matching host_ids?
What do you think?

Reply:
Currently, we are not storing the account number of user in ROS db, so we cannot filter out the hosts from ROS db depending on the incoming request. I guess as per the current scenario we are first querying the host inventory using `X-RH-IDENTITY`(account number) and then filtering out those host from ROS DB. 
We can do the reverse way as you mentioned but then first we need to store account number with the help of ROS report-processor.

Reply:
gotcha. also, in the suggestion why are we checking `len` on line 29?